pkg/workflows/interfaces/interfaces.go

```diff
@@ -3,14 +3,22 @@ package interfaces
 import (
 	"context"
 
+	"k8s.io/apimachinery/pkg/runtime"
+
 	"github.com/aws/eks-anywhere/pkg/bootstrapper"
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
 	"github.com/aws/eks-anywhere/pkg/cluster"
 	"github.com/aws/eks-anywhere/pkg/constants"
 	"github.com/aws/eks-anywhere/pkg/providers"
 	"github.com/aws/eks-anywhere/pkg/types"
 	"github.com/aws/eks-anywhere/pkg/validations"
 )
 
+// KubernetesClient allows to interact with the k8s api server.
+type KubernetesClient interface {
+	Apply(ctx context.Context, kubeconfigPath string, obj runtime.Object, opts ...kubernetes.KubectlApplyOption) error
```

Comment:
why can't we use `kubernetes.Client`?

Reply:
we can use `kubernetes.Client` instead. I think I was running into an import cycle at some point when I tried to use it at first, however, that must have been related to something else.